### ISSUE TITLE
Deprecated function: Return type of Aws\ResultPaginator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

### ISSUE THREAD

**Issue by DieterHolvoet** [aws/aws-sdk-php#2355]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though [Developer Guide](https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/aws-sdk-php/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-php-sdk) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-php/issues) and didn't find any solution
  
**Describe the bug**
When using this package with PHP 8.1, the following notice appears:
```
Deprecated function: Return type of Aws\ResultPaginator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice
```

**Version of AWS SDK for PHP?**
3.207.1

**Version of PHP (`php -v`)?**
8.1

**To Reproduce (observed behavior)**
The error appears when calling `Aws\AwsClient->getPaginator('ListObjects')`

**Expected behavior**
No deprecation notice.

**Comment by einhaus** [comment]
I got a lot of these as well when upgrading to PHP 8.1:

File: /var/www/constants/composer/vendor/aws/aws-sdk-php/src/ResultPaginator.php
Line: 104
Error: Return type of Aws\ResultPaginator::current() should either be compatible with Iterator::current(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

File: /var/www/constants/composer/vendor/aws/aws-sdk-php/src/ResultPaginator.php
Line: 114
Error: Return type of Aws\ResultPaginator::next() should either be compatible with Iterator::next(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

File: /var/www/constants/composer/vendor/aws/aws-sdk-php/src/ResultPaginator.php
Line: 109
Error: Return type of Aws\ResultPaginator::key() should either be compatible with Iterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

File: /var/www/constants/composer/vendor/aws/aws-sdk-php/src/ResultPaginator.php
Line: 119
Error: Return type of Aws\ResultPaginator::valid() should either be compatible with Iterator::valid(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

File: /var/www/constants/composer/vendor/aws/aws-sdk-php/src/ResultPaginator.php
Line: 137
Error: Return type of Aws\ResultPaginator::rewind() should either be compatible with Iterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice

**Comment by DieterHolvoet** [comment]
Will probably be fixed by https://github.com/aws/aws-sdk-php/pull/2354.

**Comment by SamRemis** [comment]
@DieterHolvoet, It should be :) can you confirm this is fixed for you as of 3.208.5?

**Comment by chekalsky** [comment]
@DieterHolvoet we also have these:

Deprecated: strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /var/task/vendor/aws/aws-sdk-php/src/Api/Validator.php on line 219

**Comment by DieterHolvoet** [comment]
> @DieterHolvoet, It should be :) can you confirm this is fixed for you as of 3.208.5?

I'm not working anymore on the project where this issue happened, @RobinHoutevelts if you want to check this on Wieni's PHP 8.1 test project, go ahead. @SamRemis no need to wait for confirmation, thanks for asking though.

**Comment by stobrien89** [comment]
It appears this has been resolved by #2354.  If anyone is still having issues, please let us know!
